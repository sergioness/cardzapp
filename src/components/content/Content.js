import React from 'react';
import { connect } from "react-redux";
import List from '../list/List';
import Searchbox from '../searchbox/Searchbox';
import Scrollable from '../scrollable/Scrollable';
import ErrorBoundry from '../errorboundry/ErrorBoundry';
import './Content.css';
import { setSearchField, requestData } from '../../actions';

const mapStateToProps = (state) => {
    return {
        ...state.searchCard,
        ...state.requestData
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        onSearchChange: (event) => {
            const action = setSearchField(event.target.value);
            return dispatch(action);
        },
        onRequestData: () => {
            return dispatch(requestData());
        }
    };
};


class Content extends React.Component {

    filter = (list, value) => {
        const valueLowerCase = value.toLowerCase();
        return list.filter((listElement) => {
            return listElement.name.toLowerCase().includes(valueLowerCase);
        });
    }

    componentDidMount() {
        this.props.onRequestData();
    }

    render() {
        const { searchField, onSearchChange, data, isPending} = this.props;
        const filteredDataList = this.filter(data, searchField);
        return isPending ?
            (<h4>Loading</h4>) :
            (<ErrorBoundry className='tc'>
                <Searchbox onSearch={onSearchChange}/>
                <Scrollable>
                    <List values={filteredDataList}/>
                </Scrollable>
            </ErrorBoundry>);
    }
};
export default connect(mapStateToProps, mapDispatchToProps)(Content);