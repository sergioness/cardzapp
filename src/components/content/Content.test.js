import Enzyme from 'enzyme';
import React from 'react';
import Content from './Content';
import toJson from 'enzyme-to-json';

let shallowWrapperContent;
describe('Component \'Content\'', () => {
  beforeEach(() => {
    const initContentProps = {
      onRequestData: jest.fn(),
      isPending: false,
      data: [],
      searchField: ''
    };
    shallowWrapperContent = Enzyme.shallow(<Content {...initContentProps}/>);
  });

  it('should shallowly match its snapshot', () => {
    expect(toJson(shallowWrapperContent, {mode: 'deep'})).toMatchSnapshot();
  });

  it('filter() should return an empty array if nothing was found', () => {
    const data = [{name: 'Mia'}, {name: 'Sean'}, {name: 'Michael'}];
    const toFind = 'John';
    const filteredData = shallowWrapperContent.instance().filter(data, toFind);
    expect(filteredData).toEqual([]);
  });

  it('filter() should return an array of objects which \'name\' property match a given one', () => {
    const data = [{name: 'Mia'}, {name: 'Sean'}, {name: 'Michael'}];
    const toFind = 'Mi';
    const filteredData = shallowWrapperContent.instance().filter(data, toFind);
    expect(filteredData).toHaveLength(2);
  });

  it('render() should render \'Loading\' heading (<h4>) if data fetching is pending', () => {
    const prevProps = shallowWrapperContent.props();
    shallowWrapperContent.setProps({...prevProps, isPending: true});
    expect(shallowWrapperContent.find('h4')).toHaveLength(1);
  });
});