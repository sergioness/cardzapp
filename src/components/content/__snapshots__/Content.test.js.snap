// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component 'Content' should shallowly match its snapshot 1`] = `
<ErrorBoundry
  className="tc"
>
  <Searchbox />
  <Scrollable>
    <List
      values={Array []}
    />
  </Scrollable>
</ErrorBoundry>
`;
